### DISCUSSION TITLE
Support to other DBMS (Roadmap item)

### DISCUSSION THREAD

**Discussion by WhistlerSupport** [fabioformosa/quartz-manager#125] [General]
May I know if this had any progress? I have looked at the quartz-manager-starter-persistence project, currently PostgreSQL is hardcoded.

1. Would the type of database be configurable through properties?
2. Would it be a simple switch statement to choose the driverClassName per database selected?
3. Does all the supported database driver need to be included in the pom dependency?

I did not understand the project enough to run a full trial, I simply drafted the laziest change I could've thought of.

    @Bean
    public DataSource quartzManagerDatasource() {
        String driverClassName;
    	
		switch (quartzDatasourceDatabaseType) {
		case "MySQL8":
		case "MySQL": //MySQL8
			driverClassName = "com.mysql.cj.jdbc.Driver";
			break;
		default:
			driverClassName = "org.postgresql.Driver";
			break;
		}
    	
        return DataSourceBuilder.create()
                .url(quartzDatasourceUrl)
                .driverClassName(driverClassName)
                .username(quartzDatasourceUser)
                .password(quartzDatasourcePassword)
                .build();
    }

added in pom
```    
<dependency>
    <groupId>mysql</groupId>
    <artifactId>mysql-connector-java</artifactId>
</dependency>
```

Not sure if this is where Author intends it to grow to

**Comment by fabioformosa** [comment]
Hi, supporting more DBMS in addition to PostgreSQL is definitely in the roadmap (Oracle, MySql, Sql Server). It will come at the beginning of 2025.
Unfortunetely it's not just a matter of a switch-statement to set the driverClassName. Quart Manager runs the liquibase migration scripts to create the tables, provided by Quartz Scheduler. Extending to other kinds of DBMS, even the migration scripts must be extended. 
New integration tests must be written, to check all DBMS against docker containers with testcontainer lib. Regarding licensed DBMS (Oracle, Mysql, Sql Server), I don't know whether there are available and unlicensed docker images for test purposes. To be investigated.
Feel free to submit a PR, you're welcome if you want to contribute developing the support for other DBMS (and the integration tests also).
